chef/cookbooks/glance/recipes/ceph.rb

```diff
@@ -0,0 +1,60 @@
+#
+# Cookbook Name:: glance
+# Recipe:: ceph
+#
+# Copyright (c) 2015 SUSE Linux GmbH.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+ceph_env_filter = " AND ceph_config_environment:ceph-config-default"
+ceph_servers = search(:node, "roles:ceph-osd#{ceph_env_filter}") || []
+if ceph_servers.length > 0
+  include_recipe "ceph::keyring"
+else
+  # If external Ceph cluster will be used,
+  # we need install ceph client packages
+  if node[:platform] == "suse" && File.exists?("/etc/ceph/ceph.client.admin.keyring")
+    package "ceph-common"
+    package "python-ceph"
+  else 
+    return
```

Comment:
It has to be implemented in that way. For cinder we have different backends and we can check what backend was enabled. For glance we can enable only one rbd backend and it has to be enabled always when it is available.